lib/output.js

```diff
@@ -74,9 +74,12 @@ function OutputStream(options) {
     var OUTPUT = "";
 
     function to_ascii(str, identifier) {
-        return str.replace(/[\u0000-\u001f\u007f-\uffff]/g, function(ch) {
-            var code = ch.charCodeAt(0).toString(16);
-            if (code.length <= 2 && !identifier) {
+        return str.replace(/[\ud800-\udbff][\udc00-\udfff]|[\u0000-\u001f\u007f-\uffff]/g, function(ch) {
```

Comment:
Hmm maybe I'll better add a `es5` flag for it this time. It's way easier to solve this problem than the other pr.

Only discussion left is if we should enable it by default or not. Users should be aware though that not everything in es6 will be backported to es5 valid code.